File: lessons/compiled_lang/cs/src/l00.exit.cs
using System;

class L00Exit
{
    const int EX_USAGE = 64;
    const int EX_OK = 0;

    static void UsageMessage(string scriptName)
    {
        Console.Error.WriteLine();
        Console.Error.WriteLine("You need to enter one or more numbers:");
        Console.Error.WriteLine();
        Console.Error.WriteLine("   Usage: " + scriptName + " [num1] [num2] [num3]...");
        Console.Error.WriteLine();
        Environment.Exit(EX_USAGE);
    }

    static void AddNums(string[] numbers)
    {
        int sum = 0;
        foreach (string num in numbers)
        {
            sum += int.Parse(num);
        }
        Console.WriteLine("The summation is: " + sum + ".");
        Environment.Exit(EX_OK);
    }

    static void Main(string[] args)
    {
        string scriptName = Environment.GetCommandLineArgs()[0];

        if (args.Length < 1)
        {
            UsageMessage(scriptName);
        }
        else
        {
            AddNums(args);
        }
    }
}

File: lessons/compiled_lang/rust/src/l00.exit.rs
use std::env;
use std::process;

const EX_USAGE: i32 = 64;
const EX_OK: i32 = 0;

fn usage_message(script_name: &str) -> ! {
    eprintln!();
    eprintln!("You need to enter one or more numbers:");
    eprintln!();
    eprintln!("   Usage: {script_name} [num1] [num2] [num3]...");
    eprintln!();
    process::exit(EX_USAGE);
}

fn add_nums(numbers: &[String]) -> ! {
    let mut sum = 0;
    for num in numbers {
        sum += num.parse::<i32>().unwrap();
    }
    println!("The summation is: {sum}.");
    process::exit(EX_OK);
}

fn main() {
    let all_args: Vec<String> = env::args().collect();
    let script_name = &all_args[0];
    let args = &all_args[1..];

    if args.is_empty() {
        usage_message(script_name);
    } else {
        add_nums(args);
    }
}
